packages/nocodb/src/lib/meta/api/userApi/initAdminFromEnv.ts

```diff
@@ -0,0 +1,209 @@
+            // get all project access belongs to the existing account
+            // and migrate to the admin account
+            const existingUserProjects = await ncMeta.metaList2(
+              null,
+              null,
+              MetaTable.PROJECT_USERS,
+              {
+                condition: { fk_user_id: existingUserWithNewEmail.id }
+              }
+            );
+
+            for (const existingUserProject of existingUserProjects) {
+              const userProject = await ProjectUser.get(
+                existingUserProject.project_id,
+                superUser.id,
+                ncMeta
+              );
+
+              // if admin user already have access to the project
+              // then update role based on the highest access level
+              if (userProject) {
+                if (
+                  rolesLevel[userProject.roles] >
+                  rolesLevel[existingUserProject.roles]
+                ) {
+                  await ProjectUser.update(
+                    userProject.project_id,
+                    superUser.id,
+                    existingUserProject.roles,
+                    ncMeta
+                  );
+                }
+              } else {
+                // if super doesn't have access then add the access
+                await ProjectUser.insert(
+                  {
+                    ...existingUserProject,
+                    fk_user_id: superUser.id
+                  },
+                  ncMeta
+                );
+              }
+              // delete the old project access entry from DB
+              await ProjectUser.delete(
+                existingUserProject.project_id,
+                existingUserProject.fk_user_id,
+                ncMeta
+              );
+            }
+
+            // delete existing user
+            ncMeta.metaDelete(
+              null,
+              null,
+              MetaTable.USERS,
+              existingUserWithNewEmail.id
+            );
+
+            // Update email and password of super admin account
+            await User.update(
+              superUser.id,
+              {
+                salt,
+                email,
+                password,
+                email_verification_token
+              },
+              ncMeta
+            );
+          } else {
+            // if email's are not different update the password and hash
+            await User.update(
+              superUser.id,
+              {
+                salt,
+                email,
+                password,
+                email_verification_token
+              },
+              ncMeta
+            );
+          }
+        } else {
+          // if email's are not different update the password and hash
+          await User.update(
+            superUser.id,
+            {
+              salt,
+              password,
+              email_verification_token
+            },
+            ncMeta
+          );
+        }
+      }
```

Comment:
When only password updated the old user can still access without re-login. If you update the mail you get token expired for same case.This might cause a security issue. We should check if there is a password change (to avoid unwanted re-login if nothing changed) and update token if needed.

Reply:
> When only password updated the old user can still access without re-login

This we can take care by updating token version to null